### PR TITLE
Enables JSX  and remove previous planning module

### PR DESCRIPTION
**Pull request by vied12**
Enables JSX

Remove previous planning module

### CODE REVIEW

**Comment by ioanpocol** [comment]
This is more like a delete not an add :)

File: scripts/index.js [hunk 1]
```diff
@@ -15,13 +15,21 @@ if (appConfig.features.useTansaProofing) {
 // since tests do not import this file.
 angular.module('superdesk.config').constant('config', appConfig);
 
+// Superdesk modules to be loaded in bootstrap process
```
Comment (gbbr) [inline scripts/index.js]: Please undo the changes in this file and remove the `features.planning` item from the configuration. It is no longer needed after https://github.com/superdesk/superdesk-client-core/pull/995.

To enable the `superdesk.planning` module, simply install it with `npm` and add this into `superdesk.config.js`:
```js
apps: ['superdesk-planning']
```

File: webpack.config.js [hunk 2]
```diff
@@ -52,26 +52,31 @@ module.exports = function makeConfig(grunt) {
...
                 {
-                    test: /\.js$/,
+                    test: /\.jsx?$/,
                     exclude: function(p) {
                         'use strict';
                         // exclude parsing node modules, but allow the 'superdesk-core'
                         // node module, because it will be used when building in the
                         // main 'superdesk' repository.
-                        return p.indexOf('node_modules') > -1 && p.indexOf('superdesk-core') < 0;
+                        var allowing = ['superdesk-core', 'superdesk-planning'];
```
Comment (gbbr) [inline webpack.config.js]: You can remove these changes too after #995

File: webpack.config.js [hunk 3]
```diff
@@ -164,6 +169,8 @@ function getDefaults(grunt) {
 
         // app features
         features: {
+            // superdesk-planning feature
```
Comment (gbbr) [inline webpack.config.js]: This too.

**Comment by vied12** [comment]
I rebased and made the changes. well done, it works like a charm !

**Comment by ioanpocol** [comment]
I think that we have to include & activate EsLint and Jest on package.json

**Comment by gbbr** [comment]
@ioanpocol JEST is not a very good testing framework and Facebook doesn't give much support... We can use our current set-up to test React with some small adjustments. Shouldn't be an issue.

**Comment by vied12** [comment]
Should we merge it or do we still "code freeze" :snowflake:  ?

File: package.json [hunk 4]
```diff
@@ -29,6 +29,7 @@
   "main": "scripts/index.js",
   "dependencies": {
     "angular-history": "git+https://github.com/decipherinc/angular-history.git#v0.8.0",
+    "superdesk-planning": "git+https://github.com/superdesk/superdesk-planning.git#1f04215",
```
Comment (gbbr) [inline package.json]: I wonder if this should be here. If yes, then why not include it straight into client-core if it is already permanently part of it? Do we really need a separate node module given that it is not optional?

It seems like this kind of defeats the whole purpose of having external modules... What do the others thinks?

Comment (vied12) [inline package.json]: I added it here because last time I tried (before your #995) it was impossible to add it at the `superdesk/superdesk` level. Ideally, in my opinion, it would be loaded there in order to be included in integration tests. But the feature must stay option-able in term of a user's point of view.

Comment (gbbr) [inline package.json]: Makes sense. Well, now it can be added there, and into that `superdesk.config.js`. So maybe in this case we should not have it here, right?

Comment (vied12) [inline package.json]: Deal !

**Comment by gbbr** [comment]
@vied12 not a big issue right now, but you might have forgotten to add `html-loader` to `package.json`

**Comment by vied12** [comment]
@gbbr that's right ! It was working for me because I have it in the sd pluggin.
here is the fix: https://github.com/superdesk/superdesk-client-core/pull/1036
